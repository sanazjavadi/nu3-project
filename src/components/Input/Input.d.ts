declare namespace IInput {
  export interface IProps {
    type: string;
    placeholder: string;
    error: string;
    className: string;
    value: any;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  }
}

export { IInput };
