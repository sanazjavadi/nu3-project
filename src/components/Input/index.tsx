import React from "react";

// InterFaces
import { IInput } from "./Input";

const Input: React.FC<IInput.IProps> = ({
  type = "text",
  placeholder,
  error,
  className,
  value,
  onChange,
  ...other
}) => {
  return (
    <div className="relative w-full">
      <input
        type={type}
        value={value}
        placeholder={placeholder}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => onChange(e)}
        id="search"
        className="block p-2.5 w-full z-20 text-sm text-gray-900 bg-gray-50 rounded-r-lg border-l-gray-100 border-l-2 border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:border-blue-500"
      />
    </div>
  );
};

Input.defaultProps = {
  type: "text",
  className: "",
};

export default Input;
