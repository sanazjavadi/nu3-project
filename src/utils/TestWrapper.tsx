import { configureStore, Store } from "@reduxjs/toolkit";
import { MemoryRouter } from "react-router-dom";
import { ComponentType, FunctionComponent } from "react";
import { Provider } from "react-redux";
import reducer from "../reducers";

export const makeStore = (): Store => {
  return configureStore({ reducer });
};

const wrapComponent = (
  Component: ComponentType | FunctionComponent | any,
  store: Store | null = null,
  props = {}
): JSX.Element => {
  return (
    <Provider store={store || makeStore()}>
      <MemoryRouter>
        <Component {...props} />
      </MemoryRouter>
    </Provider>
  );
};

export default wrapComponent;
