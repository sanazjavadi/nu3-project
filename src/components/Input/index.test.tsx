import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import TestWrapper, { makeStore } from "../../utils/TestWrapper";
import Input from ".";

const store = makeStore();

describe("Input Tests Cases", () => {
  test("AppInput: Renders", () => {
    render(TestWrapper(Input, store));
    const InputElement = screen.getByLabelText("input");
    expect(InputElement).toBeInTheDocument();
  });

  it("correctly change values when user types", async () => {
    const onChange = jest.fn();
    render(TestWrapper(Input, store, { onChange: onChange }));
    const InputElement = screen.getByLabelText("input");
    userEvent.type(InputElement, "test");

    await waitFor(() => expect(InputElement).toHaveValue("test"));
    expect(onChange).toHaveBeenCalled();
  });
});

test("AppInput: Snapshot", () => {
  const component = renderer.create(TestWrapper(Input, store));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
