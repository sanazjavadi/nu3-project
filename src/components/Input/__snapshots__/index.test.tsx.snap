// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInput: Snapshot 1`] = `
<div
  className="relative w-full"
>
  <input
    aria-label="input"
    className="appearance-none block pl-3 pr-10 py-2  w-full z-20 text-base text-gray-900 bg-gray-50 rounded-lg  border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:border-blue-500"
    id="search"
    onChange={[Function]}
    type="text"
  />
</div>
`;
